lifelines/utils/sklearnAdapter.py

```diff
@@ -0,0 +1,84 @@
+import inspect
+import pandas
+from sklearn.base import BaseEstimator, RegressorMixin, MetaEstimatorMixin, MultiOutputMixin
+from . import concordance_index
+
+
+def filterKwArgs(f, kwargs):
+    s = inspect.getfullargspec(f)
+    if s.varkw:
+        return kwargs
+    else:
+        allArgs = set()
+        if s.args:
+            allArgs |= set(s.args)
+        if s.kwonlyargs:
+            allArgs |= set(s.kwonlyargs)
+        fedArgs = set(kwargs)
+        redundantArgs = fedArgs - allArgs
+        presentArgs = fedArgs & allArgs
+        #if redundantArgs:
+        #    warnings.warn("Following args are redundant for "+str(f)+str(inspect.signature(f))+": "+repr(redundantArgs))
+        res = {k: kwargs[k] for k in presentArgs}
+        return res
+
+
+class LifelinesSKLearnAdapter(BaseEstimator, MetaEstimatorMixin, RegressorMixin):
+    __slots__ = ("fitter", "params", "yArgName", "eventArgName", "remap", "concordanceRemap")
+    def __init__(self, fitter, params, yArgName="duration_col", eventArgName="event_col", remap=None, concordanceRemap=None):
+        self.fitter = fitter
+        assert yArgName in params
+        self.params = params
+        self.yArgName = yArgName
+        self.eventArgName = eventArgName
+        if remap is None:
+            remap = {}
+        self.remap = remap
+        if concordanceRemap is None:
+            concordanceRemap = lambda sself, c: c
+        self.concordanceRemap = concordanceRemap
+
+    @property
+    def yColumn(self):
+        return self.params[self.yArgName]
+
+    @property
+    def eventColumn(self):
+        return self.params[self.eventArgName] if self.eventArgName else None
+
+    def fit(self, X, y=None, sample_weight=None):
+        """y has the default value to allow a keyword one because sklearn pass the stuff as to a keyword one"""
+        if y is not None:
+            X.insert(len(X.columns), self.yColumn, y, allow_duplicates=False)
```

Comment:
is `insert` inplace? If not, we need `X = X.insert...` here

Reply:
If it wasn't inplace, there would have been no sense to have this method. It returns `None`:

https://github.com/pandas-dev/pandas/blob/e7ad88441b2586bc7f725ecd9b354b35a3ec8c99/pandas/core/frame.py#L3437L3456